dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/DefaultAnalyticsService.java

```diff
@@ -1131,8 +1132,15 @@ private void fillGridWithAggregatedDataMap( DataQueryParams params, Grid grid, M
                 {
                     log.error( DebugUtils.getStackTrace( ex ) );
                     log.error( DebugUtils.getStackTrace( ex.getCause() ) );
-
-                    throw new RuntimeException( "Error during execution of aggregation query task", ex );
+                    
+                    if ( ex.getCause() != null && ex.getCause() instanceof RuntimeException )
+                    {
+                        throw (RuntimeException) ex.getCause(); // Throw the real exception instead of execution exception
```

Comment:
Maybe we should some log message as well? As we do in other classes? Just to be consistent. (If the exception is caught later and logged, then OK. But we should also log so it is easier to find out what is wrong just by looking into the log file)

Reply:
Yep here the exception is thrown / forwarded and hence logged. In this case, Spring wraps the exception in an ExecutionException as the method is invoked async. We are more interested in the underlying exception, hence throwing the "cause".